.editorconfig

```diff
@@ -0,0 +1,33 @@
+# EditorConfig is awesome: https://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+# Set default charset
+charset = utf-8
+indent_style = space
+indent_size = 4
+
+# Matches multiple files with brace expansion notation
+# Uncomment these when changing the default indent config
+# [*.{rs,json,py,md,markdown}]
+# indent_style = space
+# indent_size = 4
+
```

Comment:
https://github.com/editorconfig/editorconfig/wiki/Projects-Using-EditorConfig

A little confused about your comments, explain more?

Reply:
Check this: https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties

Why i comment these lines?

check num 13-14:
```
indent_style = space
indent_size = 4
```

we already have the default indent config and it's same as the comment ones.
but if we change the indent config in the future(maybe indent_size = 2), while `{rs,json,py,md,markdown}` these should be stay the old config(indent_size =4).

If you config the indent on then `[]` part, use the config. if not use the default at the header of file.

Reply:
Ok，just a template confit usage.

Reply:
It's better to uncomment these in case of changing the default config. I will fix it later